x-pack/plugins/reporting/server/lib/validate_max_content_length.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import numeral from '@elastic/numeral';
+import { get } from 'lodash';
+
+const KIBANA_MAX_SIZE_BYTES_PATH = 'xpack.reporting.csv.maxSizeBytes';
+const ES_MAX_SIZE_BYTES_PATH = 'http.max_content_length';
+
+export async function validateMaxContentLength(
+  config: any,
+  esClient: any,
+  log: (message: string) => any
+) {
+  const kibanaMaxContentBytes = config.get(KIBANA_MAX_SIZE_BYTES_PATH);
```

Comment:
It would be appropriate to just pass the `server` object to `validateMaxContentLength` and from there get the `callWithInternalUser` client function.

```
  const config = server.config();
  const { callWithInternalUser } = server.plugins.elasticsearch.getCluster('data');

  const elasticClusterSettingsResponse = await callWithInternalUser('cluster.getSettings', { includeDefaults: true });
  const { persistent, transient, defaults: defaultSettings } = elasticClusterSettingsResponse;
  const elasticClusterSettings = defaults({}, persistent, transient, defaultSettings);

  const elasticSearchMaxContent = get(elasticClusterSettings, 'http.max_content_length', '100mb');
  const elasticSearchMaxContentBytes = numeral().unformat(elasticSearchMaxContent.toUpperCase());
  const kibanaMaxContentBytes = config.get(KIBANA_MAX_SIZE_BYTES_PATH);
```

Reply:
BTW I have it using `_.defaults` because the response will come back with top-level properties to organize settings into different kinds. This merges all the settings together and checks for the value in the merged-up copy.

Reply:
Awesome, I didn't stumble upon this utility, thanks for pointing it out!